pkg/validatorfactory/factory.go

```diff
@@ -369,7 +369,14 @@ func (s *ValidatorFactory) ValidatorsForGVK(gvk schema.GroupVersionKind) (*Valid
 			continue
 		}
 
-		val := newValidatorEntry(nam, namespaced.Has(gvk), def, ssf)
+		// Try to infer the scope from paths
+		nsScoped := namespaced.Has(gvk)
+		// Check schema extensions to see if the scope was manually added
+		if scope, ok := def.Extensions.GetString("x-kubectl-validate-scope"); ok {
+			nsScoped = scope == strings.ToLower(string(apiextensions.NamespaceScoped))
```

Comment:
probably want some integration tests that this is getting correctly propagated by:

1. validating yaml for namespace-scoped resource that has no namespace specified
2. validating yaml for cluster-scoped resource that has namespace specified

Reply:
yes, i can add some tests later if it's ok